client/src/pages/DiscoveryPage.js

```diff
@@ -21,19 +43,52 @@ const useStyles = makeStyles((theme: Theme) =>
 const DiscoveryPage = () => {
     const classes = useStyles() 
     const [contests, setContests] = useState([]) 
+    const [searchString, setSearchString] = useState("")
 
-    useEffect(() => {
-        getAllContest( (data) => {
+    const fetchContests = () => {
+        getAllContest(searchString, (data) => {
```

Comment:
a question for this, if i search by say "turtle", and i clear the search box (so it'll pass an empty string ""), would the re-display the initial active contests?